swam.validation.ValidationException: cannot pop from empty operand stack.

## What I did

Extract this [core3.zip](https://github.com/satabin/swam/files/4902157/core3.zip) in the cwd and run:

```shell
$ ./millw cli.run run -d --wasi --dir=. core3
[94/128] wasi.compile 
[info] Compiling 5 Scala sources to /home/orestis/KTH/SLUMPs/swam/out/wasi/compile/dest/classes ...
[info] Done compiling.
[info] Compiling 14 Scala sources to /home/orestis/KTH/SLUMPs/swam/out/wasi/compile/dest/classes ...
[info] Done compiling.
[118/128] cli.compile 
[info] Compiling 1 Scala source to /home/orestis/KTH/SLUMPs/swam/out/cli/compile/dest/classes ...
[info] Done compiling.
[128/128] cli.run 
swam.runtime.CompileException: An error occurred during compilation
	at swam.runtime.internals.compiler.Compiler.$anonfun$compile$15(Compiler.scala:236)
	at swam.runtime.internals.compiler.Compiler.$anonfun$compile$15$adapted(Compiler.scala:236)
	at fs2.Stream$.$anonfun$handleErrorWith$1(Stream.scala:1527)
	at fs2.internal.FreeC$$anon$5.cont(FreeC.scala:65)
	at fs2.internal.FreeC$ViewL$$anon$9$$anon$10.cont(FreeC.scala:196)
	at fs2.internal.FreeC$ViewL$.mk(FreeC.scala:185)
	at fs2.internal.FreeC$ViewL$.apply(FreeC.scala:176)
	at fs2.internal.FreeC.viewL(FreeC.scala:79)
	at fs2.internal.Algebra$.go$1(Algebra.scala:178)
	at fs2.internal.Algebra$.$anonfun$compileLoop$16(Algebra.scala:311)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:142)
	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:359)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:380)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:323)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:139)
	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:359)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:380)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:323)
	at cats.effect.internals.IOShift$Tick.run(IOShift.scala:35)
	at cats.effect.internals.PoolUtils$$anon$2$$anon$3.run(PoolUtils.scala:52)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: swam.validation.ValidationException: cannot pop from empty operand stack.
	at swam.validation.Context.pop(Context.scala:104)
	at swam.validation.Context.pop(Context.scala:74)
	at swam.validation.Context.pop(Context.scala:71)
	at swam.validation.Context.$anonfun$pop$2(Context.scala:92)
	at cats.StackSafeMonad.tailRecM(StackSafeMonad.scala:15)
	at cats.StackSafeMonad.tailRecM$(StackSafeMonad.scala:15)
	at cats.effect.IOLowPriorityInstances$IOEffect.tailRecM(IO.scala:792)
	at swam.validation.Context.pop(Context.scala:87)
	at swam.validation.SpecValidator$ctraverser$.$anonfun$brTraverse$1(SpecValidator.scala:389)
	at swam.traversal.Traverser.traverse(Traverser.scala:446)
	at swam.traversal.Traverser.$anonfun$run$13(Traverser.scala:479)
	... 13 more
1 targets failed
cli.run subprocess failed
```

Like in #80, this is from the libsodium test suite and it works with wasmer

--- a/core/src/swam/syntax/pretty/package.scala
+++ b/core/src/swam/syntax/pretty/package.scala
@@ -250,10 +250,10 @@ package object pretty {
         case Unreachable => str("unreachable")
 
         case Block(tpe, is) =>
-          group(str("block") ++ space ++ tpe.pretty ++ nest(2, group(line ++ seq(line, is)) ++ line ++ str("end")))
+          group(str("block") ++ space ++ tpe.pretty ++ nest(2, group(line ++ seq(line, is))) ++ line ++ str("end"))
 
         case Loop(tpe, is) =>
-          group(str("loop") ++ space ++ tpe.pretty ++ nest(2, group(line ++ seq(line, is)) ++ line ++ str("end")))
+          group(str("loop") ++ space ++ tpe.pretty ++ nest(2, group(line ++ seq(line, is))) ++ line ++ str("end"))
 
         case If(tpe, t, e) =>
           group(

--- a/core/src/swam/validation/Context.scala
+++ b/core/src/swam/validation/Context.scala
@@ -119,7 +119,7 @@ case class Context[F[_]](types: Vector[FuncType],
   private def extractType(bt: BlockType, loop: Boolean): F[(Vector[ValType], Int)] =
     bt match {
       case BlockType.NoType         => F.pure((Vector.empty, 0))
-      case BlockType.ValueType(tpe) => F.pure((Vector(tpe), 0))
+      case BlockType.ValueType(tpe) => F.pure((if (loop) Vector.empty else Vector(tpe), 0))
       case BlockType.FunctionType(tpeIdx) =>
         types.lift(tpeIdx) match {
           case Some(FuncType(params, results)) =>

